### PR TITLE
QRCode with logo

### PR DESCRIPTION
**Pull request by Gartic99**
as asked in issue [#226](https://github.com/italia/eudi-wallet-it-python/issues/226) unused parameters were removed.
Size parameter is now passed by yaml file.
added a new library to manage qrcodes with logo in the middle.

### CODE REVIEW

File: example/satosa/pyeudiw_backend.yaml [hunk 1]
```diff
@@ -19,11 +19,9 @@ config:
       get_response: '/get-response'
 
   qrcode:
-    size: 100
+    size: 250 #px
```
Comment (peppelinux) [inline example/satosa/pyeudiw_backend.yaml:22]: we should handle a specific type, like integer here and say, even with a comment, that the value is expected in pixels

while in the js code or html template we use #px

File: example/satosa/static/js/qrcode/qr-code.orxjfzvr.js [hunk 2]
```diff
@@ -0,0 +1,5 @@
+/*! Built with http://stenciljs.com */
```
Comment (peppelinux) [inline example/satosa/static/js/qrcode/qr-code.orxjfzvr.js:1]: is this file required? it seems the copy of the previous one

File: example/satosa/pyeudiw_backend.yaml [hunk 3]
```diff
@@ -19,11 +19,9 @@ config:
...
     color: '#2B4375'
     expiration_time: 120 # seconds
-    logo_path:
```
Comment (peppelinux) [inline example/satosa/pyeudiw_backend.yaml:25]: logo_path is required, otherwise we don't allow the user to customize the logo within the qrcode

File: example/satosa/pyeudiw_backend.yaml (hunk 1 above)
Comment (peppelinux) [reply example/satosa/pyeudiw_backend.yaml:22]: if you say "c'mon Giuseppe #px is a comment" -> ok

File: example/satosa/templates/qr_code.html [hunk 4]
```diff
@@ -16,12 +16,22 @@
...
+                        position-ring-color="{{ qrcode_color }}"
+                        position-center-color="{{ qrcode_color }}",
+                        style="
+                            width: {{ qrcode_size }}px;
+                            height: {{ qrcode_size }}px;
+                            margin: 2em auto;
+                            background-color: #fff;
+                        "
+                        >
+                        <img class="icon-qr-code" src="/static/wallet-it/wallet-icon-blue.svg" slot="icon" />
```
Comment (peppelinux) [inline example/satosa/templates/qr_code.html:32]: please use the static endpoint defined in the conf and not a constant like this /static/...

then: pass the static endpoint within the template context

**Review by peppelinux — CHANGES_REQUESTED** [review]
Good job @Gartic99, some other changes are still required

File: example/satosa/templates/qr_code.html (hunk 4 above)
Comment (Gartic99) [reply example/satosa/templates/qr_code.html]: passed qr-code logo path in conf file

File: example/satosa/templates/base.html [hunk 5]
```diff
@@ -10,7 +10,7 @@
   <link rel="stylesheet" href="/static/bootstrap-italia/css/bootstrap-italia.min.css">
   <link rel="stylesheet" href="/static/css/style.css">
 
-  <script src="/static/js/qrcode.min.js"></script>
+  <script src="/static/js/qrcode/qr-code.js"></script>
   <script src="/static/js/jquery-3.7.0.min.js"></script>
   <script src="/static/bootstrap-italia/js/bootstrap-italia.bundle.min.js"></script>
```
Comment (peppelinux) [inline example/satosa/templates/base.html:15]: replace `/static` with the static web path definedi n the general configuration

File: pyeudiw/satosa/backend.py [hunk 6]
```diff
@@ -206,6 +206,8 @@ def pre_request_endpoint(self, context: Context, internal_request, **kwargs) ->
         result = self.template.qrcode_page.render(
             {
                 "qrcode_color": self.config["qrcode"]["color"],
+                "qrcode_size": self.config["qrcode"]["size"],
+                "qrcode_logo_path": self.config["qrcode"]["logo_path"],
                 "qrcode_text": res_url,
```
Comment (peppelinux) [inline pyeudiw/satosa/backend.py:211]: here, pass the static path, configured in the configuration, here:
https://github.com/italia/eudi-wallet-it-python/blob/9487df889a234bd5056040eda8614c1d9945ad5f/example/satosa/pyeudiw_backend.yaml#L7C5-L7C23

and the static storage url MUST be this

`https://localhost/static` 

and then the logo_path must be this

`logo_path: '/wallet-it/wallet-icon-blue.svg'`

**Review by peppelinux — CHANGES_REQUESTED** [review]

File: pyeudiw/satosa/backend.py [hunk 7]
```diff
@@ -206,6 +206,8 @@ def pre_request_endpoint(self, context: Context, internal_request, **kwargs) ->
...
                 "qrcode_expiration_time": self.config["qrcode"]["expiration_time"],
                 "state": state,
```
Comment (peppelinux) [inline pyeudiw/satosa/backend.py:213]: where is "static"?

https://github.com/italia/eudi-wallet-it-python/pull/230/files#diff-cca91a28aa9487e7894dee05653610c0ee47c14ba8cb54dd594ef898b0154c04R13

File: example/satosa/pyeudiw_backend.yaml [hunk 8]
```diff
@@ -19,11 +19,10 @@ config:
...
 
   qrcode:
-    size: 100
-    color: '#2B4375'
+    size: 250 # px
+    color: '#000000'  # hex
     expiration_time: 120 # seconds
-    logo_path: 
-    use_zlib: false
+    logo_path: '/static/wallet-it/wallet-icon-blue.svg' # relative to static_storage_url
```
Comment (peppelinux) [inline example/satosa/pyeudiw_backend.yaml:25]:
```suggestion
    logo_path: '/wallet-it/wallet-icon-blue.svg' # relative to static_storage_url
```

because `/static` is,  must be, part of the configured static url in the configuration

**Review by peppelinux — CHANGES_REQUESTED** [review]

File: pyeudiw/satosa/backend.py (hunk 7 above)
Comment (peppelinux) [reply pyeudiw/satosa/backend.py:213]: @Gartic99 you told me that this is loaded in the default context of the jinja2 template processor, that's good.

File: example/satosa/templates/qr_code.html [hunk 9]
```diff
@@ -28,7 +28,7 @@
                             background-color: #fff;
                         "
                         >
-                        <img class="icon-qr-code" src="{{ qrcode_logo_path }}" slot="icon" />
+                        <img class="icon-qr-code" src="{{ static+qrcode_logo_path }}" slot="icon" />
```
Comment (peppelinux) [inline example/satosa/templates/qr_code.html:31]: I don't like this solution

what happen if qrcode_logo_path doesn't start with "/"?
please provide it formatted OR add in the confgiuration validator (pydantic) the check that all the configuration to be appended to the static webendpoint MUST start with "/"

**Review by peppelinux — CHANGES_REQUESTED** [review]

**Review by peppelinux — APPROVED** [review]
